library/sketches-library/src/main/java/uk/gov/gchq/gaffer/sketches/datasketches/cardinality/function/ToHllSketch.java

```diff
@@ -44,14 +46,17 @@ public ToHllSketch(final int logK) {
         this.logK = logK;
     }
 
-    public ToHllSketch(final HllSketch hllSketch) {
-        this.hllSketch = hllSketch;
+    public ToHllSketch(final HllSketch initHllSketch) {
+        this.initHllSketch = initHllSketch;
```

Comment:
Out of curiosity could the copy be taken here instead if not null? e.g.
```java
if (initHllSketch != null) {
    this.hllSketch = initHllSketch.copy();
}
```

Reply:
I guess my thinking was someone could use the setter to overwrite it as null but that seems very unlikely. What do you think?

Reply:
I think it gets messy if another constructor is used

Reply:
Not quite sure I follow, do you mean by subclasses of this class? All I was thinking is it sounds like you want to allow an existing HllSketch to be used to init the one associated with this class but then just make sure not modify that original object so make a copy of it before using? This could be done in the constructor then the set method just does the same.

Reply:
It could be done there, but you would still need this null check in the apply in case someone uses one of the other constructors.

Reply:
You can call the other constructors from one of the overloaded ones with `this()` (I edited my comment to add an example)